test/tests/parallel/set1/spam_virus/amavis.bats

```diff
@@ -18,8 +18,10 @@ function setup_file() {
 
 function teardown_file() { _default_teardown ; }
 
-@test "${TEST_NAME_PREFIX} Amavis integration should be active" {
-  _run_in_container grep 'ANTI-SPAM-SA' /var/log/mail/mail.log
+@test "SpamAssassin integration should be active" {
+  # give Amavis just a bit of time to print out its full debug log
+  run repeat_in_container_until_success_or_timeout 5 "${CONTAINER_NAME}" grep 'ANTI-SPAM-SA' /var/log/mail/mail.log
+  assert_success
   assert_output --partial 'loaded'
   refute_output --partial 'NOT loaded'
```

Comment:
This is works too :+1: (_I noticed this locally and was going to use an existing helper method to wait for the amavis service to be ready_)

Reply:
Alternatively, instead of "grep-ing until success", grep until a string appears:

```bash
tail -F /var/log/mail/mail.log | grep --max-count 1 'ANTI-SPAM-SA'

timeout 5 tail -F /var/log/mail/mail.log | grep --max-count 1 'ANTI-SPAM-SA'
```

Reply:
> grep until a string appears

I have been refactoring a test where a similar requirement with counting was necessary and being a hassle for me to shell-fu my way through, I may want to switch to your tip here now :joy: